src/Features/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.IncrementalMemberEditAnalyzer.cs

```diff
@@ -248,7 +247,9 @@ public void UpdateDocumentWithCachedDiagnostics(Document document)
                 //      with the new diagnostics for this member node in the latest document snaphot.
                 // If we are unable to perform this incremental diagnostics update,
                 // we fallback to computing the diagnostics for the entire document.
-                if (TryGetUpdatedDocumentDiagnostics(existingData, oldMemberSpans, diagnostics, changedMember.SyntaxTree, changedMember, changedMemberId, out var updatedDiagnostics))
+                var tree = changedMember.SyntaxTree;
+                var text = tree.GetText(cancellationToken);
```

Comment:
this method is called when a method is edited.  So we have an open doc and should always have a cheap way to get the text.